db/dcrsqlite/sync.go

```diff
@@ -189,30 +202,58 @@ func (db *wiredDB) resyncDB(quit chan struct{}, blockGetter rpcutils.BlockGetter
 			db.waitChan = blockGetter.WaitForHeight(i + 1)
 		}
 
-		if i > stakeDBHeight {
-			if i != int64(db.sDB.Height()+1) {
-				panic(fmt.Sprintf("about to connect the wrong block: %d, %d", i, db.sDB.Height()))
-			}
+		stakeDBHeight = int64(db.sDB.Height())
+		blockHeight := block.Height()
+
+		// A block whose height is less than or equal the stakedb height should
+		// be ignored. Only blocks greater than the stakedb height should be
+		// considered.
+		if blockHeight > stakeDBHeight {
 			if err = db.sDB.ConnectBlock(block); err != nil {
 				return i - 1, err
 			}
 		}
 
+		// If the current block's height fetched earlier, using the notification
+		// waitChan is not greater than the current stakedb height, bypass the waitChan
+		// to fetch the blocks directly till we find a block whose height is greater
+		// than the stakedb height. For blocks whose height is less than the stakedb
+		// height, they exists in the other dbs so skip to the next iteration.
+		bypassWaitChan = (blockHeight != i)
+		if bypassWaitChan {
+			// subtraction kicks in here because of blockGetter.WaitForHeight(i + 1).
+			// Here we want consecutive blocks to be fetched without skipping any
+			// because of (i+1) in the function above when waitChan kicks in.
+			i--
+			continue
```

Comment:
`db.waitChan = blockGetter.WaitForHeight(i + 1)` only returns the notification of the current block if the previous one was processed and added to the stakedb. If by any means the previous block wasn't processed (added to stakedb) then the notification for the current block will never be sent via `db.waitChan`, the sync essentially just blocks forever.

`bypassWaitChan` is only set to true if when `i` is not equal to the new block height of the block to be sync'd. In reality this will only happen when`blockHeight > stakeDBHeight` is false thereby only running when `waitChan` expected not to send any block notifications.

If this still doesn't make sense lets have a chats about this via slack

Reply:
I guess I should remove this part because its already fixed by #720